### ISSUE TITLE
GPU Build broken on Titan

### ISSUE THREAD

**Issue by PDoakORNL** [QMCPACK/qmcpack#895]
Or perhaps misdocumented.

I get this
```
[  0%] Building NVCC (Device) object src/einspline/CMakeFiles/einspline.dir//./einspline_generated_bspline_create_cuda.cu.o
cd /lustre/atlas/proj-shared/csc249/epd/qmcpack/build/src/einspline/CMakeFiles/einspline.dir && /autofs/nccs-svm1_sw/xk6/cmake/2.8.10.2/cle4.0_gnu4.7.2/bin/cmake -E make_directory /lustre/atlas/proj-shared/csc249/epd/qmcpack/build/src/einspline/CMakeFiles/einspline.dir//.
cd /lustre/atlas/proj-shared/csc249/epd/qmcpack/build/src/einspline/CMakeFiles/einspline.dir && /autofs/nccs-svm1_sw/xk6/cmake/2.8.10.2/cle4.0_gnu4.7.2/bin/cmake -D verbose:BOOL=1 -D build_configuration:STRING=RELEASE -D generated_file:STRING=/lustre/atlas/proj-shared/csc249/epd/qmcpack/build/src/einspline/CMakeFiles/einspline.dir//./einspline_generated_bspline_create_cuda.cu.o -D generated_cubin_file:STRING=/lustre/atlas/proj-shared/csc249/epd/qmcpack/build/src/einspline/CMakeFiles/einspline.dir//./einspline_generated_bspline_create_cuda.cu.o.cubin.txt -P /lustre/atlas/proj-shared/csc249/epd/qmcpack/build/src/einspline/CMakeFiles/einspline.dir//einspline_generated_bspline_create_cuda.cu.o.cmake
-- Removing /lustre/atlas/proj-shared/csc249/epd/qmcpack/build/src/einspline/CMakeFiles/einspline.dir//./einspline_generated_bspline_create_cuda.cu.o
/autofs/nccs-svm1_sw/xk6/cmake/2.8.10.2/cle4.0_gnu4.7.2/bin/cmake -E remove /lustre/atlas/proj-shared/csc249/epd/qmcpack/build/src/einspline/CMakeFiles/einspline.dir//./einspline_generated_bspline_create_cuda.cu.o
-- Generating dependency file: /lustre/atlas/proj-shared/csc249/epd/qmcpack/build/src/einspline/CMakeFiles/einspline.dir//einspline_generated_bspline_create_cuda.cu.o.NVCC-depend
/opt/nvidia/cudatoolkit7.5/7.5.18-1.0502.10743.2.1/bin/nvcc -M -D__CUDACC__ /lustre/atlas/proj-shared/csc249/epd/qmcpack/src/einspline/bspline_create_cuda.cu -o /lustre/atlas/proj-shared/csc249/epd/qmcpack/build/src/einspline/CMakeFiles/einspline.dir//einspline_generated_bspline_create_cuda.cu.o.NVCC-depend -ccbin /opt/cray/craype/2.5.13/bin/cc -m64 -Drestrict=__restrict__ -DHAVE_CONFIG_H -DH5_USE_16_API -DUSE_REAL_STRUCT_FACTOR -DINLINE_ALL=inline -DADD_ -DMPICH_SKIP_MPICXX Drestrict=__restrict__ -DNO_CUDA_MAIN -O3 -use_fast_math -arch=sm_35 -std=c++11 -DNVCC -I/opt/nvidia/cudatoolkit7.5/7.5.18-1.0502.10743.2.1/include -I/opt/cray/fftw/3.3.4.8/interlagos/include -I/usr/include -I/opt/cray/hdf5/1.10.0.3/GNU/4.9/include -I/sw/xk6/boost/1.57.0/cle5.2up04_gnu4.9.0/include -I/lustre/atlas/proj-shared/csc249/epd/qmcpack/src -I/lustre/atlas/proj-shared/csc249/epd/qmcpack/build/src -I/lustre/atlas/proj-shared/csc249/epd/qmcpack/build/include -I/opt/nvidia/cudatoolkit7.5/7.5.18-1.0502.10743.2.1/include
nvcc fatal   : A single input file is required for a non-link phase when an outputfile is specified
CMake Error at einspline_generated_bspline_create_cuda.cu.o.cmake:203 (message):
  Error generating
  /lustre/atlas/proj-shared/csc249/epd/qmcpack/build/src/einspline/CMakeFiles/einspline.dir//./einspline_generated_bspline_create_cuda.cu.o
```

**Comment by ye-luo** [comment]
Using newer cmake?

**Comment by ye-luo** [comment]
Does config/build_olcf_titan.sh work?

**Comment by PDoakORNL** [comment]
Ah I think it was I was using the old cmake...  At least I don't see the above error anymore.  Waiting for build to finish.

**Comment by PDoakORNL** [comment]
With cmake3 it is no longer necessary to run twice for the gpu real version.  I'm checking the others now. PR coming with cmake minimum requirements guard for Cray and updated build and installation directions.

**Comment by ye-luo** [comment]
Be careful of rhea, the CI machine.

**Comment by PDoakORNL** [comment]
I will go check on there too.

**Comment by ye-luo** [comment]
If we raise cmake minimum requirement, we need to check **all** the machines and also incorporate the knowledge from miniqmc for C++ standard setting. Need to be very careful.

**Comment by PDoakORNL** [comment]
This is only for PE_cray enviroments i.e. XK.  I've checked and it doesn't effect the build on Rhea, although I have many other problems on that machine.  

I've never needed cmake twice on my laptop or on my Centos 7 builds, but I've tried not to depended on LD_LIBRARY_PATH or other magic environment variables.  My guess is a flag issue where some of the tests do not pickup -DCMAKE_xxx settings until the second time.
